server/controllers/request.js

```diff
@@ -1,30 +1,35 @@
-const Request = require("../models/Request");
-const express = require("express");
-const router = express.Router();
-
-// @desc  GET : list of requests for logged in user
-// @route GET
-
-const getRequests = async (req, res) => {
-  const requests = await Request.find({ userId: req.user.id });
-  res.send(requests);
-};
-
-// @desc  POST : Create a new request
-// @route POST
-
-const createRequest = async (req, res) => {
-  const request = await Request.create(req.body);
-  res.send(request);
-};
-
-// @desc UPDATE : Update request with approved or decline
-// @route PUT
-
-const updateRequest = async (req, res) => {
-  await Request.findByIdAndUpdate({ _id: req.params.id }, req.body);
-  const request = await Request.findOne({ _id: req.params.id });
-  res.send(request);
-};
-
-module.exports = { createRequest, getRequests, updateRequest };
+const Request = require("../models/Request");
+const express = require("express");
+const router = express.Router();
+
+// @desc  GET : list of requests for logged in user
+// @route GET
+
+const getRequests = async (req, res) => {
+  const requests = await Request.find({ userId: req.user.id });
+  res.send(requests);
+};
+
+// @desc  POST : Create a new request
+// @route POST
+
+const createRequest = async (req, res) => {
+  const request = await Request.create(req.body);
```

Comment:
This has eluded me before, but if we accept any user id here, any logged in user can create artificial requests for other owners - we should read the `req.user.id` field for the appropriate user id instead👍🏼

Reply:
Hi @ugur-akin,  Can you please explain this more?

Reply:
Hi @ugur-akin , Are you suggesting modifying req body in the controllers like below: 

```
const createRequest = async (req, res) => {
  req.body.userId = req.user.id;
const request = await Request.create(req.body);
  res.send(request);
}; 
```

Reply:
Not necessarily, modifying the body isn't necessary and could be misleading in some cases. For a quick fix you can do the following:
```
const request = await Request.create({...req.body, userId: req.user.id});
```

But ideally, we should instead destructure the body and explicitly pass the fields, e.g:
```
const { sitterId, start, end, ... } = req.body;
const userId = req.user.id;

const request = await Request.create({userId, sitterId, start, end, ...});
```

Reply:
Thanks @ugur-akin